docs/DeveloperGuide.md

```diff
@@ -1,34 +1,678 @@
+  
+  - Cons: Might significantly increase the code base as there is a need to check for the 
+    presence of the variable in user input.
+ 
+ 
+- Alternative 2: Require all values of an <code>item</code> object to be updated
+  - Pros: Will have less code to deal with having no additional parsing of input string needed.
+  
+  - Cons: Less user flexibility; user must input all parameters even if he/she does not want to update certain
+  variables.
+
+&nbsp;
+<b><a href="#developer-guide">&#129053; back to top</a></b>
+<!-- @@author -->
+
+&nbsp;
+
+<!-- @@author kokjoon97 --> 
+### 3.3 Delete feature
+#### 3.3.1 Current implementation
+
+The delete feature is implemented using a <code>DeleteCommand</code> class which extends the main
+<code>Command</code> class with an index representing that of the item to be deleted from the shopping
+list. 
+ 
+The <code>Shoco</code> class first receives user input from the <code>Ui</code> class before it creates a 
+<code>Parser</code> object and calls its <code>parseCommand</code> function to instantiate a 
+<code>DeleteCommand</code> object based on that user input.
+
+The <code>Shoco</code> class then calls the <code>execute</code> method of the <code>DeleteCommand</code> object
+which makes another call to the <code>deleteItem</code> function of the <code>ShoppingList</code> object 
+with the specified index.
+
+The following sequence diagram below shows how the delete feature works. Note the <code>Ui</code> class is
+omitted in the sequence diagram to emphasise on the other classes:
+
+![alt text](images/Delete.png)
+
+#### 3.3.2 Design considerations
+
+##### Aspect: Data structure to support the delete feature
+
+- Alternative 1 (current choice): Object-oriented style with a separate class for <code>DeleteCommand</code>
+ 
+  - Pros: Easy to add the delete feature without having to change the logic of the code much as each command object
+  is treated as a black box
+  
+  - Cons: Might significantly increase the code base with another class being added
+
+
+- Alternative 2: Implement delete feature in the <code>Shoco</code> class
+
+  - Pros: Will have less code to deal with as a new function is simply created in the <code>Shoco</code> class
+  
+  - Cons: Code becomes less organised since for every other command that we have implemented, <code>Shoco</code> class
+    simply executes those commands as black boxes, without worrying about their internal details
+
+  
+&nbsp;
+<b><a href="#developer-guide">&#129053; back to top</a></b>
+
+&nbsp;
+
+### 3.4 Find feature
+#### 3.4.1 Current implementation
+
+The find feature is implemented using a <code>FindCommand</code> class which extends the main
+<code>Command</code> class with a String representing the keyword specified by the user.
+ 
+The <code>Shoco</code> class first receives user input from the <code>Ui</code> class before it creates a 
+<code>Parser</code> object and calls its <code>parseCommand</code> function to instantiate a 
```

Comment:
Perhaps, you can use Parser#parseCommand() and for the other methods in your description?